oeps/processes/oep-0057-core-repositories.rst

```diff
@@ -0,0 +1,166 @@
+
+
+Motivation
+----------
+
+There is a desire to make the `openedx GitHub organization`_ easier to manage and understand. To do this, we must be judicious about introducing new repositories and consider removing some existing repositories. At the same time, contributors to new repositories are eager to share their work with the Open edX community and, understandably, often desire for the upstream of their repository to become reflected under the GitHub organization that bears the Open edX project's name.
+
+To best accomodate both these interests in a fair and inclusive way, this OEP aims to:
+
+* establish the purpose of the openedx GitHub organization,
+* define a transparent process for the management of the organization's set of repositories, and
+* explain how community members can prepare their repositories for inclusion in the organization.
+
+Definition of Core Repositories
+-------------------------------
+
+* We define the *Core repositories of Open edX* (or in this context, "the Core") as the entire set of public repositories hosted in the `openedx GitHub organization`_.
+
+  * Phrased another way: A repository is in the Core if and only if it is public and in the openedx GitHub organization.
+
+  * Terminology notes:
+
+    * The choice of *Core* here is inspired by the Core Contributor program, which exists in part to grant community members access to repositories in the openedx GitHub organization.
+
+    * This does not preclude other, smaller "cores" of Open edX (such as a *learning core*, *core utilities*, etc.) from existing within the Core repositories.
+
+  * **What repositories are in the Core?**
+
+    * A repository *must* be in the Core if it is directly included in `latest Open edX release`_.
+
+    * A repository *must* be in the Core if:
+
+      * it is listed as a non-optional base dependency of a repository included in the latest Open edX release *and*
+
+      * it is *not* maintained as a third-party package that is unspecific to Open edX, such as Django or Python Requests.
+
+    * In addition to the repositories that must be in the Core, a repository *may* be in the Core if:
+
+      * it is of interest to more than one entity the Open edX community (TODO: this needs better criteria); and
+
+      * it has assigned `maintainers`_ who strive to:
+
+        * review community code contributions,
+        * participate in community discussions related to the repository, and
+        * keep the repository compliant with the accepted Open edX Proposals (OEPs).
+
+* We declare that all access to write, configure, or otherwise administer Core repositories shall be granted through the `Core Contributor Program`_ and managed by its Program Administrators.
+
+* We declare that the addition or removal of repositories from the Core should by reviewed by existing Core Contributors.
```

Comment:
As far as the process goes - we've been having trouble getting more than 30-40% of any given core contributors to participate in any CC program survey or standup we arrange. I think it would be good to, like the CC nomination process, define a minimum number of votes. The process should be public, where anyone can weigh in but only CCs get votes. Time-bounded is good. Would we specifically want maintainers to weigh in (who are already CCs)? Defining a rough template of how a repo "applies" for admission may be nice: repo name, purpose, core development team, who in the community is using it, plans for future development, and plan for maintainership all come to mind.

Reply:
I think having two voting mechanisms, "affirmative consent +N" defined as requiring N positive votes and no negatives, & "lazy assent: X days" as a time-bounded default acceptance of X days given no negatives, makes sense to me! What do you think would be the best way to define these? The absolute shortest OEP ever? 😁